### PR TITLE
[docs-only] Update gateway readme

### PR DESCRIPTION
**Pull request by mmattel**
References:
#9490 (set the configured protocol transport for service metadata)
#10362 ([docs-only] ADDING 2 envvars to existing ones in settings service)
Note that 10362 is not mandatory, but if merged timely, we need to update this PR to fix the envvar names.

Based on a discussion with @butonic, the current readme text needs some sharpening and an overview what is meant with that (experimental) change.

image

### CODE REVIEW

File: services/gateway/README.md [hunk 1]
```diff
@@ -30,12 +30,20 @@ Store specific notes:
 ## Service Endpoints
 
 **IMPORTANT**\
-This functionality is currently experimental.
+This functionality is currently highly experimental and intended for testing only! There are known bugs that need to be sorted out like not closing sockets when a service ends.
```
Comment (butonic) [inline services/gateway/README.md:33]:
```suggestion
This functionality is currently highly experimental and intended for testing only! There are known bugs that need to be sorted out like not removing sockets when a service ends.
```

File: services/gateway/README.md [hunk 2]
```diff
@@ -83,15 +91,17 @@ OCM_GRPC_PROTOCOL=unix"
...
-STORAGE_GATEWAY_GRPC_ADDR="unix:/var/run/ocis/storage-system.sock"
-STORAGE_GRPC_ADDR="unix:/var/run/ocis/storage-system.sock"
 SHARING_USER_CS3_PROVIDER_ADDR="unix:/var/run/ocis/storage-system.sock"
 SHARING_USER_JSONCS3_PROVIDER_ADDR="unix:/var/run/ocis/storage-system.sock"
 SHARING_PUBLIC_CS3_PROVIDER_ADDR="unix:/var/run/ocis/storage-system.sock"
 SHARING_PUBLIC_JSONCS3_PROVIDER_ADDR="unix:/var/run/ocis/storage-system.sock"
+
+// the STORAGE_xxx envvars should (and will) be named SETTINGS_STORAGE_xxx
+STORAGE_GATEWAY_GRPC_ADDR="unix:/var/run/ocis/storage-system.sock"
+STORAGE_GRPC_ADDR="unix:/var/run/ocis/storage-system.sock"
```
Comment (butonic) [inline services/gateway/README.md:104]: reference the other issue and update this to use the corret env vars. then we can first merge the fix of the env vars and then this PR is already correct.

Comment (mmattel) [reply services/gateway/README.md:104]: Reference made, see on top.
You need to approve + merge PR: #10362 first.
This is a dev decision, therefore please press the button...
I will do the text fix commit afterwards.

Comment (mmattel) [reply services/gateway/README.md]: done, fixed

**Pull request merged**